mantidproject/mantid
# Move python Framework init to simpleapi

**Description of work.**

Traditionally, as soon as you `import mantid` the `FrameworkManager` starts and all of the plugins are loaded into the `simpleapi`. This makes it impossible to do something like `from mantid.api import MatrixWorkspace` without paying a lot for standing up the framework. This is particularly notable in the `ErrorReporter` that starts the framework again just to pull out 1 class.

This change moves all initialization to `mantid.simpleapi`.

**Report to:** [user name]/[nobody]. 

**To test:**

The following code should now not start the framework and be much quicker than before:
```python
from mantid.api import MatrixWorkspace
```

Please test that both the new Workbench and MantidPlot start correctly. All algorithms, particularly check for Python algorithms like `LoadVesuvio` are available.

Also check that `from mantid.simpleapi import *` works in a standalone Python interpreter and gives the expected functions like `LoadVesuvio` etc.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
